src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -275,13 +275,9 @@ public void setLayout(Layout layout) {
 
     public void setEntry(BibEntry newEntry) {
 
-        if (entry.isPresent() && (entry.get() != newEntry)) {
-            entry.ifPresent(e -> e.unregisterListener(this));
-        }
-        if(newEntry != null) {
-            newEntry.registerListener(this);
-        }
+        entry.filter(e -> e != newEntry).ifPresent(e -> e.unregisterListener(this));
         entry = Optional.ofNullable(newEntry);
+        entry.ifPresent(e -> e.registerListener(this));
```

Comment:
Can you please explain why this solves this issue? From a quick look, it essentially seems to be only the conversion to Optional code and no semantic changes.